stanfordkarel/karel_world.py

```diff
@@ -116,14 +116,36 @@ def __init__(self, world_file: str) -> None:
         # Save initial beeper state to enable world reset
         self.init_beepers = copy.deepcopy(self.beepers)
 
+    def get_beepers(self) -> dict[tuple[int, int], int]:
```

Comment:
Instead of adding these functions, can you change `beepers` and `colors` to be a regular dict instead of a defaultdict?

If I recall correctly, there are very few places we actually add to these dicts, so it should fairly straightforward to replace.